### PR TITLE
[Monitor][Query] Regen with DPG

### PR DESCRIPTION
**Pull request by pvaneck**
* Swagger file updated so that the metrics client and query client are generated separately.
* Removes dependency on msrest.
* Generated models were removed and hand-crafted models were adjusted to compensate.
* LogsQueryClient and MetricsQueryClient were updated to use the updated code.
* Added typing-extensions dependency.

### CODE REVIEW

**Review by kristapratico — APPROVED** [review]
Small things, LGTM!

File: sdk/monitor/azure-monitor-query/azure/monitor/query/_models.py [hunk 1]
```diff
@@ -227,9 +224,14 @@ def __init__(
...
-        return InternalLogQueryRequest(
-            id=self.id, body=self.body, headers=self.headers, workspace=self.workspace
-        )
+        return {
+            "id": self.id,
+            "body": self.body,
+            "headers": self.headers,
+            "workspace": self.workspace,
+            "path": "/query",
+            "method": "POST"
```
[the remark this answers is not in the document]
Comment (pvaneck) [reply sdk/monitor/azure-monitor-query/azure/monitor/query/_models.py:233]: Yea, they were defaulted inside the generated model: https://github.com/Azure/azure-sdk-for-python/blob/102fe91337ee5bbb51de94886aed7002bfbb286f/sdk/monitor/azure-monitor-query/azure/monitor/query/_generated/models/_models_py3.py#L57-L58

**Pull request merged**